### PR TITLE
[PP] Don't allow for num_microbatches > num_stages for single stage schedules

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #144702

There is an edge case where `Schedule1F1B` will hang when num_microbatches=1 (https://github.com/pytorch/torchtitan/issues/775). For validation it makes sense to check that the number of stages should be >= number of microbatches otherwise there will be an even larger bubble.

This can be removed when we have the single stage schedules to use an IR and updated to run with schedule runtime (issue tracker https://github.com/pytorch/pytorch/issues/144701)

cc @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by kwen2501 — APPROVED** [review]
Okay to guard against a potential bug. (Is it in theory a restriction?)

**Comment by H-Huang** [comment]
@kwen2501, yeah this is just to guard against a bug. It should not be a restriction and if we move to a different runtime then we will allow it

**Comment by cassanof** [comment]
I think this is a regression as the bug was only in the 1F1B scheduler, GPipe worked with MB < PP. It's a pretty important feature for people training very big models on long context.

**Comment by KimbingNg** [comment]
> I think this is a regression as the bug was only in the 1F1B scheduler, GPipe worked with MB < PP. It's a pretty important feature for people training very big models on long context.

Totally agree with @cassanof ! GPipe with MP < PP works fine, this should be only added to 1F1B rather than the SingleStage base class

cc @H-Huang
